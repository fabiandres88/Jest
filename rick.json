{
    "id": 1,
    "name": "Rick sanchez",
    "status": "Alive",
    "species": "Human",
    "gender": "Male"
}