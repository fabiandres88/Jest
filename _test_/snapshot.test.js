import { getCharacter } from "../snapshot";
import rick from "../rick.json";

describe("is time to the snapshots", () => {
    test("Snaphots", () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });
    test("The snapshot always will fail", () => {
        const user = {
            createdAt: new Date(),
            id: Math.floor(Math.random() * 20),
        }
        expect(user).toMatchSnapshot();
    });

    test("The snapshot has an exception", () => {
        const user = {
            updatedAt: new Date(),
            createdAt: "2021-03-02T16:39:11.339Z",
            id: 1,
            name: "Carolina Jaimes"
        }
        expect(user).toMatchSnapshot({
            updatedAt: expect.any(Date)
        });
    });
});

//jest -u to update an snapshot