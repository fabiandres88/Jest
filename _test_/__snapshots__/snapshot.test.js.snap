// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is time to the snapshots Snaphots 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`is time to the snapshots The snapshot always will fail 1`] = `
Object {
  "createdAt": 2021-03-02T16:48:30.633Z,
  "id": 4,
}
`;

exports[`is time to the snapshots The snapshot always will fail 2`] = `
Object {
  "createdAt": "2021-03-02T16:39:11.339Z",
  "id": 1,
  "name": "Carolina Jaimes",
  "updatedAt": Any<Date>,
}
`;
